smoketest/systemlimits_test.go

```diff
@@ -149,8 +154,21 @@ func TestSystemLimits(t *testing.T) {
 			}
 			setLimit := func(max int) {
 				t.Helper()
-				noErr(doQLErr(t, fmt.Sprintf(`mutation{updateConfigLimit(input:{id: %s, max: %d}){id}}`, limitID, max), parseID))
+				h.SetSystemLimit(limitID, max)
 			}
+
+			if isMultiInsert {
+				noErr(doQLErr(t, addQuery(3), parseID))
+				setLimit(2)
+				mustErr(doQLErr(t, addQuery(4), parseID))
+				noErr(doQLErr(t, addQuery(1), parseID))
+				noErr(doQLErr(t, addQuery(2), parseID))
+				mustErr(doQLErr(t, addQuery(3), parseID))
+				setLimit(-1)
+				noErr(doQLErr(t, addQuery(4), parseID))
+				return
```

Comment:
Could you add a little info here for this bit? The random numbers/func calls seem a bit magical without context.